wagtail/images/tests/test_models.py

```diff
@@ -102,6 +117,18 @@ def test_is_stored_locally(self):
     def test_is_stored_locally_with_external_storage(self):
         self.assertFalse(self.image.is_stored_locally())
 
+    @override_settings(
+        DEFAULT_FILE_STORAGE="wagtail.test.dummy_external_storage.DummyExternalStorage"
+    )
+    def test_reopen_based_on_storage_that_is_dynamically_set(self):
+        self.image.file.close()
+        # Simulate the case is witch the storage is set dynamically
+        with mock.patch.object(
+            self.image.file, "storage", AnotherDummyExternalStorage()
+        ):
```

Comment:
Question: Is there a need to mock the `storage` here if `DEFAULT_FILE_STORAGE` is set? Can one of these be removed?

Reply:
Yes, there is need to mock it because we are describing a situation that the storage is changing at runtime